src/psyclone/tests/domain/lfric/lfric_mesh_props_test.py

```diff
@@ -0,0 +1,379 @@
+
+
+# Constants
+BASE_PATH = os.path.join(
+    os.path.dirname(os.path.dirname(os.path.dirname(
+        os.path.abspath(__file__)))),
+    "test_files", "dynamo0p3")
+TEST_API = "dynamo0.3"
+
+MESH_PROPS_MDATA = '''
+module testkern_mesh_mod
+  type, extends(kernel_type) :: testkern_mesh_type
+    type(arg_type), dimension(2) :: meta_args = &
+        (/ arg_type(gh_field, gh_read, w1),     &
+           arg_type(gh_field, gh_inc, w0) /)
+    type(mesh_data_type), dimension(1) :: meta_mesh = &
+        (/ mesh_data_type(adjacent_face) /)
+     integer :: iterates_over = cells
+   contains
+     procedure, nopass :: code => testkern_mesh_code
+  end type testkern_mesh_type
+contains
+  subroutine testkern_mesh_code(a, b)
+  end subroutine testkern_mesh_code
+end module testkern_mesh_mod
+'''
+
+# Tests for parsing the metadata
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup():
+    '''Make sure that all tests here use Dynamo0.3 as API.'''
+    Config.get().api = "dynamo0.3"
+
+
+def test_mdata_parse():
+    ''' Check that we get the correct list of reference-element properties. '''
+    from psyclone.dynamo0p3 import MeshPropertiesMetaData
+    fparser.logging.disable(fparser.logging.CRITICAL)
+    code = MESH_PROPS_MDATA
+    ast = fpapi.parse(code, ignore_comments=False)
+    name = "testkern_mesh_type"
+    dkm = DynKernMetadata(ast, name=name)
+    assert dkm.mesh.properties == \
+        [MeshPropertiesMetaData.Property.ADJACENT_FACE]
+
+
+def test_mdata_invalid_property():
+    ''' Check that we raise the expected error if an unrecognised property
+    is requested. '''
+    code = MESH_PROPS_MDATA.replace("adjacent_face", "not_a_property")
+    ast = fpapi.parse(code, ignore_comments=False)
+    name = "testkern_mesh_type"
+    with pytest.raises(ParseError) as err:
+        DynKernMetadata(ast, name=name)
+    assert ("property: 'not_a_property'. Supported values are: "
+            "['ADJACENT_FACE'" in str(err.value))
+
+
+def test_mdata_wrong_arg_count():
+    ''' Check that we raise the expected error if the wrong dimension value
+    is specified for the mesh_data_type array. '''
+    code = MESH_PROPS_MDATA.replace("mesh_data_type), dimension(1)",
+                                    "mesh_data_type), dimension(4)")
+    ast = fpapi.parse(code, ignore_comments=False)
+    name = "testkern_mesh_type"
+    with pytest.raises(ParseError) as err:
+        DynKernMetadata(ast, name=name)
+    assert "'meta_mesh' metadata, the number of args" in str(err.value)
+
+
+def test_mdata_wrong_name():
+    ''' Check that we raise the expected error if the array holding properties
+    of the mesh is given the wrong name. '''
+    code = MESH_PROPS_MDATA.replace("meta_mesh =", "meta_meshes =")
+    ast = fpapi.parse(code, ignore_comments=False)
+    name = "testkern_mesh_type"
+    with pytest.raises(ParseError) as err:
+        DynKernMetadata(ast, name=name)
+    assert ("No kernel metadata with type name 'meta_mesh' found"
```

Comment:
The trouble is, we don't know what the incorrect name is - only that we can't find the supplied name. I've improved the error message to say which variable name it is that it can't find and to print the part of the meta-data that its searching.